cmd/systray/systray.go

```diff
@@ -387,15 +410,15 @@ func watchIPNBus(ctx context.Context) {
 	}
 }
 
-func watchIPNBusInner(ctx context.Context) error {
-	watcher, err := localClient.WatchIPNBus(ctx, ipn.NotifyInitialState|ipn.NotifyNoPrivateKeys)
+func (menu *Menu) watchIPNBusInner() error {
+	watcher, err := menu.lc.WatchIPNBus(menu.bgCtx, ipn.NotifyNoPrivateKeys)
 	if err != nil {
 		return fmt.Errorf("watching ipn bus: %w", err)
 	}
 	defer watcher.Close()
 	for {
 		select {
-		case <-ctx.Done():
+		case <-menu.bgCtx.Done():
```

Comment:
or just add a Done method maybe?

Reply:
a `menu.Done()` you mean? If so, then I have to worry about whether bgCtx has been set at that point, right? If not, I guess I just return a nil chan, which can never be read from?